### PR TITLE
MLPAB-2170 Send notification when voucher confirms

### PR DESCRIPTION
**Pull request by hawx**

### CODE REVIEW

File: cypress/e2e/voucher/confirm-your-name.cy.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 describe('Confirm your name', () => {
     beforeEach(() => {
-        cy.visit('/fixtures/voucher?redirect=/confirm-your-name');
+        cy.visit('/fixtures/voucher?redirect=/confirm-your-name&progress=skip');
```
Comment (acsauk) [inline cypress/e2e/voucher/confirm-your-name.cy.js:3]: Was this supposed to be `progress=` to match up with the changes in voucher fixtures?

File: internal/lpastore/lpa.go [hunk 2]
```diff
@@ -495,7 +495,7 @@ func FromDonorProvidedDetails(l *donordata.Provided) *lpadata.Lpa {
 		Restrictions:                             l.Restrictions,
 		WhenCanTheLpaBeUsed:                      l.WhenCanTheLpaBeUsed,
 		LifeSustainingTreatmentOption:            l.LifeSustainingTreatmentOption,
-		SignedAt:                                 l.SignedAt,
+		SignedAt:                                 l.WitnessedByCertificateProviderAt,
```
Comment (acsauk) [inline internal/lpastore/lpa.go:498]: I understand this change but it bothers me that there's some potential for a drift between signing and witnessing dates here if a CP fails to witness after the donor signs that could be confusing. To keep in sync, do we need to add `WitnessedByCertificateProviderAt` to the Lpa store data? (not as part of this ticket)

**Review by acsauk — COMMENTED** [review]
One potential change on a fixture link

File: cypress/e2e/voucher/confirm-your-name.cy.js (hunk 1 above)
Comment (hawx) [reply cypress/e2e/voucher/confirm-your-name.cy.js:3]: nope, purposefully using one that doesn't exist so it doesn't do the sharecode sending

File: internal/lpastore/lpa.go (hunk 2 above)
Comment (hawx) [reply internal/lpastore/lpa.go:498]: potentially, but we'll never drive logic off our `SignedAt` date as it isn't _really_ signed at that point

File: cypress/e2e/voucher/confirm-your-name.cy.js (hunk 1 above)
Comment (hawx) [reply cypress/e2e/voucher/confirm-your-name.cy.js]: I was using one that didn't exist to skip the behaviour of it being unset, but I've rewritten how it works to maybe be more obvious

**Review by acsauk — APPROVED** [review]
:shipit:

**Pull request merged**